File: Sources/BiangBiangUI/Protocols/AudioProvider.swift
import SwiftUI

@MainActor
public protocol AudioProvider: AnyObject {
    /// Speak/stream the given text/result. `languageCode` is the variant's BCP-47 code.
    func play(text: String, languageCode: String?)
    func stop()
    var isPlaying: Bool { get }
}

File: android/biangbiang-ui/src/main/java/dev/veeso/biangbiangui/protocols/AudioProvider.kt
package dev.veeso.biangbiangui.protocols

/**
 * Plays/streams audio for transliteration results.
 * Mirrors iOS `AudioProvider`.
 */
interface AudioProvider {
    /** Speak/stream the given text/result. [languageCode] is the variant's BCP-47 code. */
    fun play(text: String, languageCode: String?)

    fun stop()

    val isPlaying: Boolean
}
